### PR TITLE
fix(select/radio): Add support for options with boolean type

### PR DESCRIPTION
**Pull request by remotecom**
Description
Adding validation for boolean values in Radio input type.

[Internal ticket](https://linear.app/remote/issue/CXP-742)

Problem
Currently the validation for Radio inputs was set up for either `string` or `number` types. When using `boolean` values the validation defaulted to `string` and errored even if it was correct.

Solution
Adding a validation option for boolean types.

### CODE REVIEW

File: src/tests/createHeadlessForm.test.js
Comment (remotecom) [inline src/tests/createHeadlessForm.test.js:815]: Added test for `number` type, since it was missing

Comment (sandrina-p) [reply src/tests/createHeadlessForm.test.js:815]: Perfect @Arcombe-Remote, thanks!

File: src/tests/createHeadlessForm.test.js
Comment (sandrina-p) [inline src/tests/createHeadlessForm.test.js:811]: suggestion: Would you agree this is better?
```suggestion
        type: schemaInputTypeRadioBoolean.properties.over_18.type,
```

File: package.json
Comment (sandrina-p) [inline package.json:3]: This seems good to go! Let's just wait for the internal CI/CD to be green and then we can revert this dev version, merge this and release the final version!

File: src/tests/createHeadlessForm.test.js
Comment (sandrina-p) [inline src/tests/createHeadlessForm.test.js:810]: bug: Something is still wrong in this code because if I pass `"true"` or `"false"`, it's still valid when it should not be.

For example:

```js
  const validateForm = (vals) => friendlyError(handleValidation(vals));

  expect(validateForm({over_18: 'true'})).toEqual({
    over_18: 'The option "true" is not valid.',
  });
```

Comment (Arcombe-Remote) [reply src/tests/createHeadlessForm.test.js:810]: Looks like yup casts 'true' to `true` when using boolean()

Comment (Arcombe-Remote) [reply src/tests/createHeadlessForm.test.js:810]: `
      validateForm({ over_18: 'true' });
      validateForm({ over_18: 'false' });
      expect(validateForm({over_18: 'maybe'})).toEqual({
        over_18: 'The option "maybe" is not valid.',
      });
`

Screenshot 2024-08-19 at 21 19 02
Screenshot 2024-08-19 at 21 19 08

Comment (sandrina-p) [reply src/tests/createHeadlessForm.test.js:812]: Saw the latest commit, looks good! Release a new dev version with it, and once our internal MR has green tests, then we can merge this and release the final version.

Comment (sandrina-p) [reply src/tests/createHeadlessForm.test.js:812]: @Arcombe-Remote , I'm optimistic the internal MR will pass, so feel free to rever the version back to what it is in main (package and package-lock) and I'll aprove this right away :)

Comment (sandrina-p) [reply src/tests/createHeadlessForm.test.js:812]: @brennj could you re-approved this MR? It was already reviewed/approved by me, but I had to push a commit to revert to the main version. That way I can unblock @Arcombe-Remote when he starts his day tomorrow.

Comment (brennj) [reply src/tests/createHeadlessForm.test.js:812]: course

**Review by brennj — APPROVED** [review]

File: src/tests/createHeadlessForm.test.js
Comment (sandrina-p) [inline src/tests/createHeadlessForm.test.js:11]: FYI @Arcombe-Remote , I've renamed the PR title to follow conventional commits :) fix(select/radio): Add support for options with boolean type (#89)

Comment (sandrina-p) [reply src/tests/createHeadlessForm.test.js:11]: 🎉 Version `0.11.3-beta.0` published!

**Pull request merged**
